pkg/csi/service/common/authmanager.go

```diff
@@ -326,16 +339,103 @@ func getDatastoresWithBlockVolumePrivs(ctx context.Context, vc *vsphere.VirtualC
 }
 
 // Creates a map of vsan datastores to file service status (enabled/disabled).
+// Contains only datastores belonging to clusters with vSAN FS enabled and Host.Config.Storage privileges.
 func getDsToFileServiceEnabledMap(ctx context.Context, vc *vsphere.VirtualCenter) (map[string]bool, error) {
 	log := logger.GetLogger(ctx)
 	log.Debugf("Computing the cluster to file service status (enabled/disabled) map.")
-	datacenters, err := vc.ListDatacenters(ctx)
+
+	//Get clusters with vSAN FS enabled and privileges
+	clustersWithPrivAndFS, err := getFSEnabledClustersWithPriv(ctx, vc)
 	if err != nil {
-		log.Errorf("failed to find datacenters from VC: %+v, Error: %+v", vc.Config.Host, err)
+		log.Errorf("failed to get the file service enabled clusters with privileges. error: %+v", err)
 		return nil, err
 	}
 
 	dsToFileServiceEnabledMap := make(map[string]bool)
+	pc := property.DefaultCollector(vc.Client.Client)
+	properties := []string{"info", "summary"}
+	for _, cluster := range clustersWithPrivAndFS {
+		var dsList []vim25types.ManagedObjectReference
+		var dsMoList []mo.Datastore
+		datastores, err := cluster.Datastores(ctx)
```

Comment:
retreiving datastore managed object from properties is repeated in `getDsToFileServiceEnabledMap` and `getFSEnabledClusterToDsURLsMap`. Can you create a separate function to retrieve datastore managed objects so that both these functions can use it.

This way that piece of code is not repeated.

Reply:
Makes sense. Done.